.travis.yml

```diff
@@ -7,32 +7,41 @@ matrix:
     - os: windows
       env: BUILD_CONFIGURATION=Release
     - os: linux
-      dist: trusty
-      group: travis_lts
-      sudo: required
-      env: BUILD_CONFIGURATION=Debug
+      compiler: gcc
+      addons: &gcc9
+        apt:
+          sources: ['ubuntu-toolchain-r-test']
+          packages: ['g++-9', 'gcc-9']
+      env: GCCVER='9' CXX_COMPILER=g++-${GCCVER} CXX=${CXX_COMPILER} C_COMPILER=gcc-${GCCVER} CC=${C_COMPILER} AR=gcc-ar-${GCCVER} RANLIB=gcc-ranlib-${GCCVER} NM=gcc-nm-${GCCVER} BUILD_CONFIGURATION=Release
     - os: linux
-      dist: trusty
-      group: travis_lts
-      sudo: required
-      env: BUILD_CONFIGURATION=Release
+      compiler: gcc
+      addons: &gcc9
+        apt:
+          sources: ['ubuntu-toolchain-r-test']
+          packages: ['g++-9', 'gcc-9', 'doxygen', 'doxygen-doc', 'doxygen-latex', 'doxygen-gui', 'graphviz']
+      env: GCCVER='9' CXX_COMPILER=g++-${GCCVER} CXX=${CXX_COMPILER} C_COMPILER=gcc-${GCCVER} CC=${C_COMPILER} AR=gcc-ar-${GCCVER} RANLIB=gcc-ranlib-${GCCVER} NM=gcc-nm-${GCCVER} BUILD_CONFIGURATION=Debug
     - os: osx
       osx_image: xcode9.1
       env: BUILD_CONFIGURATION=Debug
     - os: osx
       osx_image: xcode9.1
       env: BUILD_CONFIGURATION=Release
 
-before_install:
-  - . .travis/before-install.sh
-
-install:
-  - . .travis/install.sh
-
 script:
   - mkdir build
   - cd build
   - . ../.travis/build.sh
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then cd .. ; mkdir -p docs/html ; doxygen 2>&1 > doxygen.log || cat doxygen.log ; mkdir website ; cp -r docs/* website/ ; fi
```

Comment:
What do you think about moving this line into `build.sh`?

Reply:
Another question: is it possible to check the branch name here? Cause this code will call docs generation in every branch on linux twice (debug && release)

Reply:
build.sh could be put into the `.travis` folder without any guarantee that it works somewhere else. But ok, lets keep the current approach.

>we can add a test to only generate in one of the cases
It would be good to do :)

I believe `generate_and_deploy_doxygen.sh` can be removed too?